### PR TITLE
TEColorParameter and custom ui

### PR DESCRIPTION
**Pull request by jkbelcher**
New TEColorParameter gives patterns 3 ways to consume color, wrapped in one convenient parameter:
- Solid color
- Two solid colors
- Gradient

Open Belcher_color.lxp to see example patterns of each.

It’s easiest to use with a midi controller attached… a knob mapped to this parameter turns the subparameter “Offset”.  Offset shifts the solid colors as well as the gradient.  If you don’t have a midi controller just click on the parameter and drag Offset with the mouse.

Includes a custom UI for TEPerformancePattern (knobs visually line up with the MidiFighterTwister), and a custom overlay UI for the new color parameter.

Added [the beginning of] a TE-specific Gradient pattern and a TESolidPattern as examples.  And modified @raldi 's Checkers pattern for a two-color example.

### CODE REVIEW

**Comment by jkbelcher** [comment]
This makes it easy for patterns to stay within the palette and brings the TE gradients documented by @jvyduna to the surface for easy consumption.  And keeps things runtime adjustable by the user.

File: src/main/java/titanicsend/pattern/mike/Checkers.java [hunk 1]
```diff
@@ -47,9 +39,10 @@ public Checkers(LX lx) {
     }
   }
 
-  public void run(double deltaMs) {
-    int color0 = this.oddColor.calcColor();
-    int color1 = this.evenColor.calcColor();
+  @Override
+  protected void runTEAudioPattern(double deltaMs) {
+    int color0 = getCurrentColor();
+    int color1 = getCurrentColor2();
```
[the remark this answers is not in the document]
Comment (jkbelcher) [reply src/main/java/titanicsend/pattern/mike/Checkers.java:45]: Ha I was going for minimal lines changed but yes that's silly.

[the remark this answers is not in the document]
Comment (jkbelcher) [reply src/main/java/titanicsend/pattern/mike/Checkers.java]: Fixed.

**Review by zranger1 — CHANGES_REQUESTED** [review]
Working fine here!

File: src/main/java/titanicsend/pattern/TEPerformancePattern.java [hunk 2]
```diff
@@ -15,6 +19,293 @@
...
+            .setDescription("For a solid color: Whether to use global TE palette (preferred), or a static color unique to this pattern");
+
+        public final CompoundParameter color2offset =
+            new CompoundParameter("C2Offset", 0.5);
+
+        // GRADIENT
+
+        public final EnumParameter<TEGradient> gradient =
+            new EnumParameter<TEGradient>("Gradient", TEGradient.FULL_PALETTE)
+            .setDescription("Which TEGradient to use. Full_Pallete=entire, Foreground=Primary-Secondary, Primary=Primary-BackgroundPrimary, Secondary=Secondary-BackgroundSecondary");
```
[the remark this answers is not in the document]
Comment (jkbelcher) [reply src/main/java/titanicsend/pattern/TEPerformancePattern.java]: Thx, fixed.

**Review by zranger1 — APPROVED** [review]
Approved!   Merge as soon as we get the Pixelblaze framework PR in!

**Pull request merged**